docs/user_guide/building_blocks/categories.md

```diff
@@ -1,3 +1,495 @@
+def my_categorization_producer_setup(self: Producer, reqs: dict, inputs: dict, reader_targets: InsertableDict) -> None:
+    reader_targets["mlcolumns"] = inputs["ml"]["mlcolumns"]
+
+
+# This *init* function is used to add categories as part of the Producer init. This means that
+# your config will always contain this category as long as you're running some task where this
+# particular Producer has been requested.
+@my_categorization_producer.init
+def my_categorization_producer_init(self: Producer) -> None:
+    # ensure that categories are only added to the config once
+    tag = f"{my_categorization_producer_init.__name__}_called"
+    if self.config_inst.has_tag(tag):
+        return
+
+    # add categories to config inst
+    self.config_inst.add_category(
+        name="my_category",
+        id=12345,
+        selection="my_categorizer",
+    )
+```
+
+
+## Helper functions for categories
+
+### get_events_from_categories
+
+To obtain the events of a certain category (or categories) of an akward array `events` that contains the
+`category_ids` column, you can use the  {py:func}`~columnflow.util.get_events_from_categories` function.
+
+
+```python
+from columnflow.util import get_events_from_cateogries
+
+# you can pass a string of a category in combination with the config inst
+selected_events = get_events_from_cateogries(events, "my_category", config_inst)
+
+# it is also possible to directly pass a category. In that case, no config inst is needed
+selected_events = get_events_from_cateogries(events, config_inst.get_category("my_category"))
+
+# it is also possible to pass a list of categories. In that case, all events
+# that belong to one of the requested categories are selected
+selected_events = get_events_from_cateogries(events, ["cat1", "cat2"], config_inst)
+```
+
+This function automatically consideres category ids of all leaf categories from the requested
+categories. This is especially helpful when your category contains multiple leaf categories.
+
+
+## Key points (TL;DR)
+
+- Categories are {external+order:py:class}`order.category.Category` instances defined in the config.
+- We can add categories to each other; the "smallest unit" of category is referred to as "leaf category".
+- We build each category by combining all of it's leaf categories (both in columns and in histograms).
+- Leaf categories need to define one or multiple {py:class}`~columnflow.categorization.Categorizer`,
+which is a function that defines whether or not an event belongs in this category.
+- The `category_ids` column is produced via the
+{py:class}`~columnflow.production.categories.category_ids` Producer.
+- Make sure that for each category, all of it's leaf categories are defined orthogonal to prevent double counting.
+- Groups of categories can be combined via {py:func}`~columnflow.config_util.create_category_combinations`.
+- Make sure that each group of categories is inclusive; otherwise, all combined categories will not be inclusive aswell.
```

Comment:
Not sure if we actually need this point - sometimes, you don't want to be inclusive in the first place, right?

Reply:
That's correct, however, I think it's important to mention the point that the category combinations also have an impact on the parent categories (e.g. my 2jet category also categorizes based on the number of leptons because I added a layer of lepton categories that is not inclusive).

Alternative proposal:

```
- Combining of categories can impact your parent categories; this can be prevented by defining
your categories such that each group of categories is inclusive
```